src/rust/ide/ast/impl/src/lib.rs

```diff
@@ -231,12 +231,18 @@ impl Ast {
         self
     }
 
-    /// Wraps given shape with an optional ID into Ast.
+    /// Wraps given shape with ID into Ast (uses random new ID if None).
     /// Length will ba automatically calculated based on Shape.
     pub fn new<S:Into<Shape<Ast>>>(shape:S, id:Option<Id>) -> Ast {
-        let shape: Shape<Ast> = shape.into();
+        let shape  = shape.into();
+        let id     = id.unwrap_or_else(Id::new_v4);
         let length = shape.len();
-        Ast::new_with_length(shape,id,length)
+        Ast::from_ast_id_len(shape,Some(id),length)
+    }
+
+    pub fn module(shape:Module<Ast>) -> Ast {
+        let length = shape.len();
+        Ast::from_ast_id_len(shape.into(),None,length)
```

Comment:
This is not consistent with other constructors named after shapes. They take (usually generic) input that can be used to construct shape and then wrap it in Ast.
Here it should be something like IntoIterator<Item=BlockLine<Ast>> or sth. 
That'd also actually simplify the usage of this method.